python/ray/ml/utils/gbdt_utils.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, Type, Any
+import warnings
+
+from ray import tune
+from ray.util.annotations import DeveloperAPI
+from ray.ml.trainer import Trainer
+from ray.ml.checkpoint import Checkpoint
+from ray.tune import Trainable
+from ray.tune.utils.placement_groups import PlacementGroupFactory
+from ray.ml.constants import PREPROCESSOR_KEY
+
+import xgboost_ray
+
+PARAMS_KEY = "params"
+DMATRIX_PARAMS_KEY = "dmatrix_params"
+
+
+@DeveloperAPI
+class GBDTTrainer(Trainer):
+    """Common logic for XGBoost-Ray and LightGBM-Ray."""
+
+    _use_lightgbm: bool = False
+
+    def _validate_dmatrix_params(self, config: Dict[str, Any], datasets: Dict) -> None:
+        if DMATRIX_PARAMS_KEY in config:
+            for key in config[DMATRIX_PARAMS_KEY]:
+                if key not in datasets:
+                    raise KeyError(
+                        f"'{DMATRIX_PARAMS_KEY}' dict contains key '{key}' "
+                        f"which is not present in `datasets`."
+                    )
+
+    def _get_dmatrices(
+        self, dmatrix_params: Dict[str, Any]
+    ) -> Dict[str, xgboost_ray.RayDMatrix]:
+        dmatrix_class = xgboost_ray.RayDMatrix
```

Comment:
Should this just be a class level variable that can be overwritten by XGBoostTrainer and LightGBMTrainer?

Reply:
makes sense!